src/frontend/components/VideoPlayer/index.tsx

```diff
@@ -73,22 +68,62 @@ const VideoPlayer = ({
           if (defaultVolume !== undefined) {
             videoPlayer.volume(Math.min(1, Math.max(0, defaultVolume)));
           }
+          timedTextTracks
+            .filter((track) => track.is_ready_to_show)
+            .filter((track) =>
+              [
+                timedTextMode.CLOSED_CAPTIONING,
+                timedTextMode.SUBTITLE,
+              ].includes(track.mode),
+            )
+            .forEach((track) => {
+              if (playerRef.current) {
+                playerRef.current.addTrack(track, languages);
+              }
+            });
         },
       );
 
-      document.dispatchEvent(
-        new CustomEvent('marsha_player_created', {
-          detail: {
-            videoNode: videoNodeRef.current!,
-          },
-        }),
-      );
-
       /** Make sure to destroy the player on unmount. */
-      return () => createdPlayer && createdPlayer.destroy();
+      return () => playerRef.current && playerRef.current.destroy();
     }
   }, []);
 
+  useEffect(() => {
+    if (video && video.urls && video.urls.manifests && playerRef.current) {
```

Comment:
```suggestion
    if (video?.urls?.manifests && playerRef.current) {
```

Reply:
I think @RVassili won't accept that

Reply:
as it is only syntactic sugar I see no reason to not use it.